src/codegen/generators/expression_generator.rs

```diff
@@ -710,11 +749,13 @@ impl<'a, 'b> ExpressionCodeGenerator<'a, 'b> {
                         .find_input_parameter(function_name, index as u32)
                         .and_then(|var| self.index.find_effective_type(var.get_type_name()))
                 })
-                .unwrap();
+                .ok_or_else(|| {
+                    CompileError::unknown_type(parameter.get_type_name(), left.get_location())
+                })?;
             //load the function prameter
             let pointer_to_param = builder
                 .build_struct_gep(parameter_struct, index as u32, "")
-                .unwrap();
+                .expect(INTERNAL_LLVM_ERROR);
```

Comment:
Isn't this hiding the original issue now?
Is there a way to incorporate the original err with ours?

Reply:
the build_struct_gep has a strange signature:
`build_struct_gep(&self, ptr: PointerValue<'ctx>, index: u32, name: &str) -> Result<PointerValue<'ctx>, ()>`

it returns a () for an error ... so there is no real message :-(